src/composables/websocket.ts

```diff
@@ -0,0 +1,188 @@
+import { v4 as genUuid } from 'uuid';
+import { ref, watch } from '@vue/composition-api';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { WEB_SOCKET_CHANNEL_TYPES, handleUnknownError } from '../popup/utils';
+import type {
+  IDefaultComposableOptions,
+  IMiddlewareWebSocketSubscriptionMessage,
+  INetwork,
+  WebSocketChannelTypes,
+} from '../types';
+import { useGetter } from './vuex';
+
+let wsClient : WebSocket;
+const isWsConnected = ref(false);
+// eslint-disable-next-line no-unused-vars
+const subscribers: Record<WebSocketChannelTypes, Record<string, (payload: any) => void>> = {
+  Transactions: {},
+  MicroBlocks: {},
+  KeyBlocks: {},
+};
+
+const channels: Record<WebSocketChannelTypes, boolean> = {
+  Transactions: false,
+  MicroBlocks: false,
+  KeyBlocks: false,
+};
+
+// eslint-disable-next-line no-unused-vars
+export function useWebSocket({ store }: IDefaultComposableOptions) {
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+
+  /**
+   * This function is used to subscribe to a specific WebSocket channel.
+   * @param {WebSocketChannelTypes} channelName - The name of the channel to subscribe to
+   */
+  function subscribeToWebSocketChannel(channelName: WebSocketChannelTypes) {
+    if (
+      !channelName
+      || channels[channelName]
+      || !isWsConnected.value
+    ) {
+      return;
+    }
+
+    const message: IMiddlewareWebSocketSubscriptionMessage = {
+      op: 'Subscribe',
+      payload: channelName,
+    };
+
+    wsClient.send(JSON.stringify(message));
+    channels[channelName] = true;
+  }
+
+  /**
+   * Handle websocket message event.
+   * @param {MessageEvent} message - The message event object.
+   */
+  function handleWebsocketMessage(message: MessageEvent) {
+    if (!message.data) {
+      return;
+    }
+    try {
+      const data = camelcaseKeysDeep(JSON.parse(message.data));
+      if (
+        !data.subscription
+        || !subscribers[data.subscription as WebSocketChannelTypes]
+      ) {
+        return;
+      }
+      // Call all subscribers for the channel
+      Object.values(subscribers[data.subscription as WebSocketChannelTypes]).forEach(
+        (subscriberCb: any) => subscriberCb(data.payload),
+      );
+    } catch (error) {
+      handleUnknownError(error);
+    }
+  }
+
+  /**
+   * Subscribes to websocket channels.
+   */
+  function subscribeToWebSocketChannels() {
+    // Add an event listener for messages
+    wsClient.addEventListener('message', handleWebsocketMessage);
+
+    // Subscribe to all the channels
+    (Object.keys(channels) as WebSocketChannelTypes[])
```

Comment:
This line does the same as the line 99 but is written in a different way. I suggest to maintain consistency.